diff --git a/lib/parse.js b/lib/parse.js
index 3a6988e629f52aee6609dc5ec290a5e9ff47cecb..2b0cae34b6346081b2bc8f8c01f194dd98e7e633 100644
--- a/lib/parse.js
+++ b/lib/parse.js
@@ -48,6 +48,7 @@ const parse = (input, options = {}) => {
   let index = 0;
   let depth = 0;
   let value;
+  const maxBraceDepth = 100;
 
   /**
    * Helpers
@@ -196,6 +197,9 @@ const parse = (input, options = {}) => {
 
     if (value === CHAR_LEFT_CURLY_BRACE) {
       depth++;
+      if (depth > maxBraceDepth) {
+        throw new SyntaxError(`Input exceeds max brace nesting depth (${maxBraceDepth})`);
+      }
 
       const dollar = prev.value && prev.value.slice(-1) === '$' || block.dollar === true;
       const brace = {
